kolibri_explore_plugin/assets/src/views/DownloadPage.vue

```diff
@@ -141,7 +141,10 @@
         return this.setUpdateInterval();
       },
       onConfirm() {
-        this.$router.go(0);
+        // Reload the page so the app fetches new data from the backend. Upon
+        // reloading, it will redirect to the landing page.
+        // <https://github.com/endlessm/kolibri-explore-plugin/issues/687>
+        window.location.reload();
```

Comment:
This is fine as is, but why reload this page, get a redirect and load another page? Why not just load the desired page directly?

Reply:
That redirect happens inside the [download route handler](https://github.com/endlessm/kolibri-explore-plugin/blob/v6.46.0/kolibri_explore_plugin/assets/src/routes/index.js#L20-L25) in the Vue code. It does a couple of async things in [`decideDownload()`](https://github.com/endlessm/kolibri-explore-plugin/blob/v6.46.0/kolibri_explore_plugin/assets/src/modules/topicsRoot/handlers.js#L183-L188) to figure out there is not actually a download happening, but it's all pretty trivial stuff happening in the client - it even does `router.replace()`, so the download page page gets dropped from the history stack.